lib/config/validation.ts

```diff
@@ -818,139 +818,141 @@ async function validateGlobalConfig(
   warnings: ValidationMessage[],
   currentPath: string | undefined,
 ): Promise<void> {
-  if (type === 'string') {
-    if (is.string(val)) {
-      if (
-        key === 'onboardingConfigFileName' &&
-        !configFileNames.includes(val)
-      ) {
-        warnings.push({
-          topic: 'Configuration Error',
-          message: `Invalid value \`${val}\` for \`${currentPath}\`. The allowed values are ${configFileNames.join(', ')}.`,
-        });
-      } else if (
-        key === 'repositoryCache' &&
-        !['enabled', 'disabled', 'reset'].includes(val)
-      ) {
-        warnings.push({
-          topic: 'Configuration Error',
-          message: `Invalid value \`${val}\` for \`${currentPath}\`. The allowed values are ${['enabled', 'disabled', 'reset'].join(', ')}.`,
-        });
-      } else if (
-        key === 'dryRun' &&
-        !['extract', 'lookup', 'full'].includes(val)
-      ) {
+  if (val !== null) {
```

Comment:
can we warn if the default value is not `null`

Reply:
Will need to create a new record that stores the default values, similar to `optionTypes`, `optionParents`. Is that okay?

Reply:
sure, let's do it in separate PR